Open Judge System/Web/OJS.Web/Controllers/TempController.cs

```diff
@@ -161,5 +168,77 @@ public ActionResult DeleteDuplicatedParticipantsInSameContest()
 
             return this.Content($"Done! Deleted Participants count: {deletedParticipantsCount}");
         }
+
+        public async Task<ActionResult> EqualizeUserIdFromSulsAndJudgeByUserName(string userName)
+        {
+            var userInfoResponse = await this.httpRequester.GetAsync<ExternalUserInfoModel>(
+                new { userName },
+                string.Format(UrlConstants.GetUserInfoByUsernameApiFormat, Settings.SulsPlatformBaseUrl),
+                Settings.ApiKey);
+
+            if (!userInfoResponse.IsSuccess || userInfoResponse.Data == null)
+            {
+                return this.Content($"Cannot get user info from SoftUni Platform for user \"{userName}\"");
+            }
+
+            var correctUserId = userInfoResponse.Data.Id;
+
+            var user = this.Data.Users.GetByUsername(userName);
+
+            if (user == null)
+            {
+                return this.Content($"Cannot find user with UserName: \"{userName}\" in the database");
+            }
+
+            if (user.Id == correctUserId)
+            {
+                return this.Content($"User \"{userName}\" has the same Id as in SoftUni Platform");
+            }
+
+            var tempUserIdToStoreParticipants = this.Data.Users.GetByUsername("gogo4ds")?.Id;
+
+            if (string.IsNullOrEmpty(tempUserIdToStoreParticipants))
+            {
+                return this.Content("Invalid temp UserId to store participants");
+            }
+
+            var participantsForUser = this.participantsData.GetAllByUser(user.Id);
```

Comment:
Shouldn't this be `participationsByUser`?

Reply:
It can be both